draft-ietf-tls-exported-authenticator.md

```diff
@@ -142,70 +175,104 @@ CertificateVerify
+algorithm used should be chosen from the "signature_algorithms" extension
+of the ClientHello used in the connection handshake.
+
+The signature is computed using the over the concatenation of:
 
 * A string that consists of octet 32 (0x20) repeated 64 times
 * The context string "Exported Authenticator" (which is not NULL-terminated)
 * A single 0 byte which serves as the separator
-* The value
-Hash(Handshake Context || Certificate) where Hash is the hash function for the handshake.
+* If the authenticator request is present, the value
+Hash(Handshake Context || authenticator request || Certificate)
+* If an authenticator request is not present, the value
+Hash(Handshake Context || Certificate)
+where
+Hash is the hash function for the handshake.
 
 Finished
 : A HMAC over the value
-Hash(Handshake Context || Certificate || CertificateVerify) where Hash
-is the hash function for the handshake, and the HMAC is computed
-using the hash function from the handshake and the Finished MAC Key as a key.
+Hash(Handshake Context || Certificate || CertificateVerify) if an authenticator
+is present, or Hash(Handshake Context || authenticator request ||
+Certificate || CertificateVerify) where Hash is the hash function for
+the handshake, and the HMAC is computed using the hash function from
+the handshake and the Finished MAC Key as a key.
 {:br}
 
-The certificates used in the Certificate message MUST conform to the requirements
-of a Certificate message in the version of TLS negotiated.  This is
-described in Section 4.2.3 of {{!TLS13}} and
-Sections 7.4.2 and 7.4.6 of {{!RFC5246}}.  Alternative certificate formats such as
-{{!RFC7250}} Raw Public Keys are not supported.
-
-The exported authenticator message is the concatenation of messages:
+The certificates chosen in the Certificate message MUST conform to the requirements
+of a Certificate message in the version of TLS negotiated.  If an authenticator
+request is present, the signature algorithms used to choose the algorithm are taken
+from the "signature_algorithms" in the from the authenticator.  If there is
+no authenticator request, the signature algorithms are chosen from the "signature_algorithms"
+extension from the ClientHello used in the connection.  This is described in
+Section 4.2.3 of {{!TLS13}} and Sections 7.4.2 and 7.4.6 of {{!RFC5246}}.
+Alternative certificate formats such as {{!RFC7250}} Raw Public Keys
+are not supported.  The "server_name" {{!RFC6066}}, "certificate_authorities"
+(Section 4.2.4. of {{!TLS13=I-D.ietf-tls-tls13}}), or "oid_filters"
+(Section 4.2.5. of {{!TLS13=I-D.ietf-tls-tls13}}) extensions are used to guide
+certificate selection, with the extensions provided in the authenticator request
+taking precedence over the extensions provided in the connection handshake.
+
+An extension MUST only be present in a Certificate message if the
+extension is present in the authenticator request if present.  If the authenticator
```

Comment:
"present in X if present" is slightly confusing wording.  Maybe reverse the order:

> If an authenticator request was provided, the Certificate message MUST contain only extensions present in the authenticator request.  Otherwise, the Certificate message MUST contain only extensions present in the ClientHello....